pkg/prowgen/prowgen.go

```diff
@@ -236,6 +236,19 @@ func generatePostsubmitsForPromotion(jobBaseBuilderFactory func() *prowJobBaseBu
 	for _, opt := range options {
 		opt(opts)
 	}
+
+	generatePromotionPostsubmit := func(jobBaseGen *prowJobBaseBuilder) *prowconfig.Postsubmit {
```

Comment:
I am not sure what this change is about. The only difference in the current implementation should be the cluster field.

Reply:
This factor out some code common to:
```golang
for _, arch := range architectures {
    ...
}
```
and
```golang
info.Config.MultiArch {
    ...
}
```
It's the same as before, just avoiding copy-paste.

Reply:
Let's don't refactor anything and focus only on the changes that are related to the PR's purpose.